app/services/preserved_object_handler.rb

```diff
@@ -72,7 +72,7 @@ def check_existence
           pres_copy = PreservedCopy.find_by!(preserved_object: pres_object, endpoint: endpoint) if pres_object
 
           if pres_copy.version != pres_object.current_version
-            handler_results.add_result(PreservedObjectHandlerResults::PC_PO_VERSION_MISMATCH, pres_copy.class.name)
+            handler_results.add_result(PreservedObjectHandlerResults::PC_PO_VERSION_MISMATCH, { pc_version: pres_copy.version, po_version: pres_object.current_version })
             raise ActiveRecord::Rollback, 'PO current_version != PC version'
```

Comment:
this looks great!  Can you stick the versions in the ActiveRecord rollback message, too?